### PR TITLE
Add fallback to storing monitoring data dir instead of prometheus snapshot

### PR DESCRIPTION
**Pull request by aleksbykov**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [ ] ~~All new and existing unit tests passed (`hydra unit-tests`)~~
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/utils.py [hunk 1]
```diff
@@ -888,3 +888,7 @@ def inner(*args, **kwargs):
                     cls_self.version,
                     cls_self.__class__.__name__))
         return inner
+
+
+class PrometheusSnapshotErrorException(Exception):
```
Comment (bentsi) [inline sdcm/utils.py]: we need a better place to store all the exceptions which SCT throws... currently better to put this in the nearest place in the code that throws it

File: sdcm/cluster.py [hunk 2]
```diff
@@ -3610,18 +3610,23 @@ def collect_logs(self, storing_dir):
...
         result = ps.create_snapshot()
-        self.log.debug(result)
         if "success" in result['status']:
             snapshot_dir = os.path.join(self.monitoring_data_dir,
                                         "snapshots",
                                         result['data']['name'])
+            return snapshot_dir
         else:
-            return ""
+            raise PrometheusSnapshotErrorException
```
Comment (bentsi) [inline sdcm/cluster.py]: `raise PrometheusSnapshotErrorException()`

File: sdcm/cluster.py [hunk 3]
```diff
@@ -3610,18 +3610,23 @@ def collect_logs(self, storing_dir):
 
         return storing_dir
 
-    def get_prometheus_snapshot(self, node):
-        # avoid cyclic-decencies between cluster and db_stats
-
+    @retrying(n=3, sleep_time=2, allowed_exceptions=(PrometheusSnapshotErrorException, ), message='Create prometheus snapshot')
```
Comment (bentsi) [inline sdcm/cluster.py]: Creating Prometheus snapshot

Comment (bentsi) [reply sdcm/cluster.py]: sleep 10 sec

File: sdcm/cluster.py [hunk 4]
```diff
@@ -3610,18 +3610,23 @@ def collect_logs(self, storing_dir):
...
+            return snapshot_dir
         else:
-            return ""
+            raise PrometheusSnapshotErrorException
+
+    def get_prometheus_snapshot(self, node):
+        try:
+            snapshot_dir = self.request_prometheus_snapshot(node)
+        except PrometheusSnapshotErrorException:
+            snapshot_dir = self.monitoring_data_dir
```
Comment (bentsi) [inline sdcm/cluster.py]: add a warning to log that you are failing back to the data dir

File: sdcm/cluster.py (hunk 2 above)
Comment (bentsi) [reply sdcm/cluster.py]: better `raise PrometheusSnapshotErrorException(result)`

File: sdcm/utils.py (hunk 1 above)
Comment (aleksbykov) [reply sdcm/utils.py]: Thought about separate file.

Comment (aleksbykov) [reply sdcm/utils.py]: fixed

File: sdcm/cluster.py (hunk 3 above)
Comment (aleksbykov) [reply sdcm/cluster.py]: fixed

File: sdcm/cluster.py (hunk 4 above)
Comment (aleksbykov) [reply sdcm/cluster.py]: fixed

File: sdcm/cluster.py (hunk 2 above)
Comment (aleksbykov) [reply sdcm/cluster.py]: fixed

**Pull request merged**